test/ngAria/ariaSpec.js

```diff
@@ -954,6 +954,59 @@ describe('$aria', function() {
       expect(clickEvents).toEqual(['div(true)', 'li(true)', 'div(true)', 'li(true)']);
     });
 
+    it('should trigger a click in browsers that provide `event.which` instead of `event.keyCode`',
+      function() {
+        compileElement(
+          '<section>' +
+            '<div ng-click="onClick($event)"></div>' +
+            '<ul><li ng-click="onClick($event)"></li></ul>' +
+          '</section>');
+
+        var divElement = element.find('div');
+        var liElement = element.find('li');
+
+        divElement.triggerHandler({type: 'keydown', which: 13});
+        liElement.triggerHandler({type: 'keydown', which: 13});
+        divElement.triggerHandler({type: 'keydown', which: 32});
+        liElement.triggerHandler({type: 'keydown', which: 32});
+
+        expect(clickEvents).toEqual(['div(true)', 'li(true)', 'div(true)', 'li(true)']);
+      }
+    );
+
+    it('should not prevent default keyboard action if the target element has editable content',
+      function() {
+        compileElement(
+          '<section>' +
+            '<div ng-click="onClick($event)" ng-attr-contenteditable="{{ edit }}"></div>' +
+            '<ul ng-click="onClick($event)"><li ng-attr-contenteditable="{{ edit }}"></li></ul>' +
```

Comment:
Why not also check static `contenteditable` attributes, as well as dynamically setting the attribute via `ng-attr...`? Or is this already covered elsewhere?

Reply:
This is just an easy way to add/remove the attribute. Since we are not really checking the attribute itself (but rather checking the `isContentEditable` property which is automatically updated by the browser based on the attribute), I felt this is fine.